### PR TITLE
Reward Inflation Upgrade Handler

### PR DESCRIPTION
**Pull request by AustinoBombino**

### CODE REVIEW

File: app/setup_handlers.go [hunk 1]
```diff
@@ -4,13 +4,29 @@ import (
...
+		if !ok {
+			panic("failed to convert mint amount")
+		}
+		mintCoins := sdk.NewCoins(sdk.NewCoin(app.StakingKeeper.BondDenom(ctx), mintAmount))
+		err := app.BankKeeper.MintCoins(ctx, dispensationtypes.ModuleName, mintCoins)
+		if err != nil {
+			panic(err)
+		}
+		// err = app.BankKeeper.SendCoinsFromModuleToAccount(ctx, dispensationtypes.ModuleName, address, mintCoins) // TODO: get destination address
+		// if err != nil {
```
Comment (marshsif) [inline app/setup_handlers.go]: sif1ct2s3t8u2kffjpaekhtngzv6yc4vm97xajqyl3
the destination address, which is the same one we've used for distributions in the past

File: app/setup_handlers.go [hunk 2]
```diff
@@ -4,13 +4,33 @@ import (
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	dispensationtypes "github.com/Sifchain/sifnode/x/dispensation/types"
 )
 
-const upgradeName = "0.9.12"
+const upgradeName = "Reward-Mint"
```
Comment (intl-man) [inline app/setup_handlers.go:11]: We need to set this as the version (e.g.: 0.9.13).

**Comment by banshee** [comment]
This shouldn't involve develop.  Needs to be a branch from master.

**Review by banshee — CHANGES_REQUESTED** [review]
This can't be branched from develop.

File: app/setup_handlers.go [hunk 3]
```diff
@@ -4,13 +4,33 @@ import (
...
 
 func SetupHandlers(app *SifchainApp) {
 	app.UpgradeKeeper.SetUpgradeHandler(upgradeName, func(ctx sdk.Context, plan types.Plan) {
 		app.Logger().Info("Running upgrade handler for " + upgradeName)
+
+		mintAmount, ok := sdk.NewIntFromString("200000000000000000000000000")
+		if !ok {
+			panic("failed to convert mint amount")
+		}
+		mintCoins := sdk.NewCoins(sdk.NewCoin(app.StakingKeeper.BondDenom(ctx), mintAmount))
```
Comment (banshee) [inline app/setup_handlers.go:21]: We need a check to make sure the mint only happens on a single release.  If someone did a hotfix without removing this code, I think we'd mint another batch.

**Review by banshee — CHANGES_REQUESTED** [review]

**Comment by AustinoBombino** [comment]
closed to branch off of master:
https://github.com/Sifchain/sifnode/pull/2172

**Review by iverc — COMMENTED** [review]
@AustinoBombino we should add a version check to only trigger the logic on a specific version (0.9.14) and discard it otherwise.

**Pull request closed without merging**